### DISCUSSION TITLE
Automated port pr publishing

### DISCUSSION THREAD

**Discussion by mathisloge** [microsoft/vcpkg#17128] [Ideas]
So i was wondering if there could be some automated way to publish ports directly from a github action

This would significantly minimize the effort for library maintainers if an action could be run on each release that would automatically open a PR in vcpkg and adjust the appropriate versions, etc. 

It would be best if there was a pre-built action for this. The target repository should be customizable to also include private registiries

**Comment by AugP** [comment]
This is a really interesting suggestion and I am curious where we can take it. I'll tag our engineering team to share their thoughts on the feasibility.

**Comment by mathisloge** [comment]
@AugP is there some progress on this?

I've some real world projects which are wondering, why there isn't a automated workflow, for at least normal updating. 

https://github.com/Dobiasd/frugally-deep/pull/277#issuecomment-1329222919
https://github.com/commschamp/comms/issues/1#issuecomment-1315962787

**Comment by Dobiasd** [comment]
To give some additional motivation, JFrog ConanCenter does this, and this is a much more pleasant experience from the point of view of a library maintainer. :wink:

**Reply by autoantwort** [reply]
Do you have a link?

**Reply by Dobiasd** [reply]
Ah, sorry. I think I meant conda-forge.

**Reply by autoantwort** [reply]
I think creating a GithubAction for that is not so hard, but you need a personal access token to create the PR in the GithubAction

**Reply by Dobiasd** [reply]
> Ah, sorry. I think I meant conda-forge.

With all the C++-package managers, that one "needs" to support, I sometimes forget which ones I need to do something with and which ones I do not, when publishing a new version of one of my two main libraries ([FunctionalPlus](https://github.com/Dobiasd/FunctionalPlus) and [frugally-deep](https://github.com/Dobiasd/frugally-deep)).

Writing all steps down for all package managers also does not help that much in my case, because I don't release that often, and after a year or so, it's well possible that too many things already have changed. "Publishing" really is the main pain point for me in maintaining my libraries over the years. :woozy_face:

And thus, it would be really awesome if all package managers could automatically "pull" the latest versions instead of me needing to manually "push" every time. :pray:

**Reply by Dobiasd** [reply]
> I think creating a GithubAction for that is not so hard, but you need a personal access token to create the PR in the GithubAction

Yeah, makes sense. :+1:
But why repeat that effort `number_of_library` times (i.e., in each lib), instead of solving it only once (i.e., in the package manager)?

**Reply by autoantwort** [reply]
I think this is the goal. There is also https://github.com/microsoft/vcpkg-tool/pull/717#issuecomment-1263834985. 
But this still needs manual maintenance (for example when a lib needs a new dependency).

**Reply by Dobiasd** [reply]
> I think this is the goal.

Nice, looking forward to it! :rocket:

> But this still needs manual maintenance (for example when a lib needs a new dependency).

Yeah, that's reasonable, I think. :+1:
